crates/fuel-core/src/combined_database.rs

```diff
@@ -61,7 +75,10 @@ pub struct CombinedDatabase {
     relayer: Database<Relayer>,
     gas_price: Database<GasPriceDatabase>,
     compression: Database<CompressionDatabase>,
-    block_aggregation: Database<BlockAggregatorDatabase>,
+    #[cfg(feature = "rpc")]
+    block_aggregation_storage: Database<BlockAggregatorDatabaseStorage>,
+    #[cfg(feature = "rpc")]
+    block_aggregation_s3: Database<BlockAggregatorDatabaseS3>,
```

Comment:
We should ahve only one instance of database per service. If we change the type of storage after a restart, it is fine to throw an error that the storage must be cleaned.